src/nixos.rs

```diff
@@ -63,13 +76,18 @@ impl OsRebuildArgs {
             update(&self.common.installable, self.update_args.update_input)?;
         }
 
-        let hostname = match &self.hostname {
+        let system_hostname = hostname::get()
+            .context("Failed to get hostname")?
+            .to_str()
+            .unwrap()
```

Comment:
I'm not really a huge fan of `unwrap()`. Bit of a nit maybe, but I think we can do better. For example:

```rs
.and_then(|h| h.to_str().map(String::from).ok_or_else(|| anyhow!("Failed to convert hostname to string")))?;
```